hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsClient.java

```diff
@@ -761,6 +763,8 @@ public AbfsRestOperation append(final String path, final byte[] buffer,
       requestHeaders.add(new AbfsHttpHeader(USER_AGENT, userAgentRetry));
     }
 
+    addCheckSumHeaderForWrite(requestHeaders, buffer);
```

Comment:
Lets send in the offset and length here as well. Although from `AbfsOutputStream`, we are setting offset 0. But in future it may be changed. Better to keep a check of that in the calculation of md5.

Reply:
Agree with this point.